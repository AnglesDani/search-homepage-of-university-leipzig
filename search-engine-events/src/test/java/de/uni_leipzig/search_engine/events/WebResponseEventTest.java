package de.uni_leipzig.search_engine.events;

import java.util.HashMap;
import java.util.Map;

import javax.servlet.http.HttpServletRequest;

import org.approvaltests.Approvals;
import org.junit.Test;
import org.mockito.Mockito;
import org.springframework.web.context.request.ServletRequestAttributes;

public class WebResponseEventTest
{
	@Test
	public void checkWebResponseEventCreationWithNullAsSessionId()
	{
		Map<String, String> header = new HashMap<>();
		header.put("key-1", "value-1");
		header.put("key-2", "value-2");
		
		HttpServletRequest request = TestUtils.mockServletRequest("http://www.bla.de", header);
		ServletRequestAttributes attributes = new ServletRequestAttributes(request);
		
		attributes = Mockito.spy(attributes);
		Mockito.doReturn(null).when(attributes).getSessionId();
		
		Approvals.verify(new WebResponseEvent(attributes));
	}
	
	@Test
	public void checkWebResponseEventCreationWithNonEmptySessionId()
	{
		Map<String, String> header = new HashMap<>();
		header.put("key-1", "value-1");
		header.put("key-2", "value-2");
		
		HttpServletRequest request = TestUtils.mockServletRequest("http://www.bla.de", header);
		ServletRequestAttributes attributes = new ServletRequestAttributes(request);
		
		attributes = Mockito.spy(attributes);
		Mockito.doReturn("MySessionId").when(attributes).getSessionId();
		
		Approvals.verify(new WebResponseEvent(attributes));
	}
}
