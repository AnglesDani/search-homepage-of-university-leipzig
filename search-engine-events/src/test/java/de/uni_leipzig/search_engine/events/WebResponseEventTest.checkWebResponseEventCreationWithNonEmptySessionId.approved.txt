WebResponseEvent(request=Request(headers={key-1=value-1, key-2=value-2}, requestUrl=http://www.bla.de, remoteAddr=0.0.0.0, remoteHost=localhost, remotePort=1234), clientId=MySessionId, responseModel=null)